shipyard_blueprints/motherduck/shipyard_motherduck/motherduck.py

```diff
@@ -0,0 +1,202 @@
+import duckdb
+from shipyard_templates import ShipyardLogger, Database, ExitCodeException
+from shipyard_templates.database import (
+    FetchError,
+    UploadError,
+    ConnectionError,
+    QueryError,
+)
+from shipyard_motherduck.errors import (
+    TempTableCreationFailed,
+    TempTableInsertFailed,
+    TempTableDropFailed,
+)
+from typing import Optional
+
+logger = ShipyardLogger().get_logger()
+
+
+class MotherDuckClient(Database):
+    def __init__(self, token: str, database: Optional[str] = None):
+        self.token = token
+        self.database = database
+        if database:
```

Comment:
Just worth noting: In the cli you're explicitly passing None where empty string but that logic seems to already be handled here so it is not needed in the scripts